CONTRIBUTING.md

```diff
@@ -25,13 +26,76 @@ _As contributors and maintainers of this project, and in the interest of fosteri
 
 ## Getting Started
 
-Dev guides:
-
-- [Contribution Guide]
-- [MacOS Dev Guide]
-- [Windows Dev Guide]
-
-General resources for contributors:
+### Forking Kustomize and Working Locally
+The Kustomize project uses a "Fork and Pull" workflow that is standard to GitHub. In git terms, your personal fork is referred to as the "origin" and the actual project's git repository is called "upstream". To keep your personal branch (origin) up to date with the project (upstream), it must be configured within your local working copy.
+
+### Create a fork in GitHub
+1. Visit https://github.com/kubernetes-sigs/kustomize
+2. Click the `Fork` button on the top right
+
+### Clone the repository
+```bash
+# Clone your repository fork from the previous step
+export user=<your github profile name>
+git clone --recurse-submodules git@github.com:$user/kustomize.git
+cd kustomize
+
+# Configure upstream
+git remote add upstream https://github.com/kubernetes-sigs/kustomize
+git remote set-url --push upstream no_push
+
+# Review git configuration
+git remote -v
+```
+
+### Create a working branch
+You will need to periodically fetch changes from the `upstream` repository to keep your working branch in sync.
+```bash
```

Comment:
Below is an example of creating a new branch to work on.

Maybe we can add example of syncing working branches that were already created on upstream repository changes.

Reply:
+1. I think it'd be a good idea to split the part that speaks to syncing a fork from the part that creates a new working branch.